business_requirement_deliverable/models/business.py

```diff
@@ -80,11 +83,31 @@ def product_uom_change(self):
             unit_price = self.product_id.standard_price
         if self.uom_id and unit_price:
             if self.uom_id.uom_type == 'bigger':
-                self.unit_price = unit_price * self.uom_id.factor_inv
+                if self.product_id.uom_id.uom_type == 'bigger':
```

Comment:
Please do rewrite these code.
you can check the function of model: `product.uom`: **_compute_qty**.